src/etc/inc/pfsense-utils.inc

```diff
@@ -366,6 +366,26 @@ function gen_user_settings_fields(&$section, $pconfig) {
 	gen_pagenamefirst_field($section, $pconfig['pagenamefirst']);
 }
 
+/****f* pfsense-utils/gen_requirestatefilter_field
+ * NAME
+ *   gen_requirestatefilter_field
+ * INPUTS
+ *   Pointer to section object
+ *   Initial value for the field
+ * RESULT
+ *   no return value, section object is updated
+ ******/
+function gen_requirestatefilter_field(&$section, $value) {
+	$section->addInput(new Form_Checkbox(
+		'requirestatefilter',
+		'Require State Filter',
+		'Do not display state table without a filter',
+		$value
+	))->setHelp('By default, the entire state table is displayed when entering '.
+		'Diagnostics > States. This option requires a filter to be entered '.
+		'before the states are displayed. Useful for systems with large state tables.');
+}
```

Comment:
Ditto but several other fields on the page are also functions so it looks like it was done to match the style already in use on the page.